client/my-sites/plan-features/index.jsx

```diff
@@ -90,6 +92,15 @@ export default connect( ( state, ownProps ) => {
 		rawPrice: getPlanRawPrice( state, planProductId /**, get from abtest **/ ),
 		planConstantObj: plansList[ ownProps.plan ],
 		billingTimeFrame: get( planObject, 'bill_period_label', '' ),
+		onUpgradeClick: ownProps.plan === PLAN_FREE ? noop : () => {
+			const selectedSiteSlug = getSiteSlug( state, selectedSiteId );
+			addItemToCart( { product_slug: plansList[ ownProps.plan ].getStoreSlug() } );
+			const checkoutPath = ownProps.selectedFeature
+				? `/checkout/features/${ownProps.selectedFeature}/${ selectedSiteSlug }`
+				: `/checkout/${ selectedSiteSlug }`;
+
+			page( checkoutPath );
```

Comment:
It might be worthwhile to decompose this for clarity, eg in pseudocode:
```javascript
planUpgradeClick() {
    addPlanToCart();
    navigateToCheckout();
}
```
Though I'm not sure it's super worth it right now, since we'll hopefully be converting the cart to Redux in the near future.